### PR TITLE
feat: replace custom-metadata-manager

### LINKED ISSUE #2436: Improve accessibility of contributor fields in Book Info
> When arriving at an input field that is populated with contributor names, the contributors are not announced to assistive technologies, and the input is announced as blank even when values have been populated. It was possible to reach the contributor names in read mode, however tabbing through the inputs some users may still assume the input is blank. It is also difficult to switch into read mode with NVDA because the arrow keys open the suggestions menu, making it even more challenging to read the content within these fields.

### LINKED ISSUE #2749: Replace Custom Metadata manager
As the Pressbooks product manager, I want users to be able to use custom field imputs on the Pressbooks book info page (which is itself a custom post type) without relying on old and deprecated code. 

I'd prefer replacing the very old Custom Metadata manager code we've copied into symbionts: https://github.com/pressbooks/pressbooks/blob/dev/symbionts/custom-metadata/custom_metadata.php) and use to create custom field inputs: https://github.com/pressbooks/pressbooks/blob/dev/inc/admin/metaboxes/namespace.php with a solution that is accessible and maintainable by our core team.

DoD: 
- [ ] preserve existing functionality & test coverage 
- [ ] remove all usage of the Custom Metadata code that's in pressbooks/pressbooks' symbionts folder from all repos

### PR DESCRIPTION
**Pull request by greatislander**
- Resolves #2436
- Resolves #2749
- Partial resolution of #2976 

### Features
 
- [x] Metabox class
- [x] Field classes
	- [x] Field
	- [x] Date
	- [x] Text
	- [x] URL
	- [x] Textarea
	- [x] Wysiwyg
	- [x] Select
	- [x] Taxonomy Select
	- [x] Checkbox
- [x] Book Information page
	- [x] General Information
	- [x] Subjects
	- [x] BISAC Subjects
	- [x] Institutions
	- [x] Copyright
	- [x] About
	- [x] Additional Catalog Information
- [x] Front Matter/Chapter/Back Matter Metadata
- [x] Part Visibility

### CODE REVIEW

**Comment by greatislander** [comment]
PHP CodeSniffer appears to be failing because of this: https://github.com/squizlabs/PHP_CodeSniffer/issues/3641

It could be solved by upgrading to WordPress Coding Standards 3.x but HumanMade Coding Standards is still locked to 2.3 🙈

**Comment by greatislander** [comment]
Ready for review. Requires changes to Pressbooks BISAC as well (see: https://github.com/pressbooks/pressbooks-bisac/issues/46)

**Comment by greatislander** [comment]
Requires these changes too: https://github.com/pressbooks/pressbooks-bisac/pull/47

File: templates/metaboxes/fields/wysiwyg.blade.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div class="form-field">
+	<label for="{{ $field->id ?? $field->name }}">
+		{{ $field->label }}
+	</label>
+	@php(wp_editor($value ?? '', $field->name, \Pressbooks\Editor\metadata_manager_default_editor_args(['textarea_rows' => $field->rows ?? 20])))
```
Comment (fdalcin) [inline templates/metaboxes/fields/wysiwyg.blade.php]:
```suggestion
	@php(wp_editor($field->value ?? '', $field->name, \Pressbooks\Editor\metadata_manager_default_editor_args(['textarea_rows' => $field->rows ?? 20])))
```

File: templates/metaboxes/fields/textarea.blade.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	<textarea
+		id="{{ $field->id ?? $field->name }}"
+		name="{{ $field->name }}"
+		type="{{ $field->type ?? 'text' }}"
+		@if($field->disabled) disabled @endif
+		@if($field->readonly) readonly @endif
+		@if(isset($field->description)) aria-describedby="{{ $field->id ?? $field->name . '-description' }}" @endif
+	>
+		{{ $value ?? '' }}
+	</textarea>
```
Comment (fdalcin) [inline templates/metaboxes/fields/textarea.blade.php]:
```suggestion
	<textarea
		id="{{ $field->id ?? $field->name }}"
		name="{{ $field->name }}"
		type="{{ $field->type ?? 'text' }}"
		@if($field->disabled) disabled @endif
		@if($field->readonly) readonly @endif
		@if(isset($field->description)) aria-describedby="{{ $field->id ?? $field->name . '-description' }}" @endif
	>{{ $field->value ?? '' }}</textarea>
```

**Review by fdalcin — CHANGES_REQUESTED** [review]
Hi @greatislander, thank you for your time working on this PR. I really like the new metabox approach.

While reviewing it we've detected a couple of minor issues where content was not being displayed properly, so I added some comments with suggestions.

We've also noticed a weird behaviour with the wysiwyg field. As you can see in the image below, when changing to **Text** mode the wysiwyg's layout renders weirdly. I'm not entirely sure about the cause, but am assuming some styling might have been removed by accident.

image

**Comment by greatislander** [comment]
@fdalcin Hi! I've addressed the issues, thanks for your review. Let me know if you encounter any other problems while testing.

**Comment by fdalcin** [comment]
Hi @greatislander, I merged https://github.com/pressbooks/pressbooks/pull/3541 and now we have a couple of conflicts here. Would you mind resolving the conflicts before I can finish my review?

**Review by fdalcin — APPROVED** [review]
LGTM, tested locally and the issues were resolved. Thank you, @greatislander.

**Comment by arzola** [comment]
LGTM

**Pull request merged**